### ISSUE TITLE
Reuse or get rid of TrailDBEvent 

### ISSUE THREAD

**Issue by buremba** [aholyoke/traildb-java#13]
Currently, `TrailDBCursor.next` creates a new TrailDBEvent and sets the fields of it with the internal values of `tdb_event`. `TrailDBCursor.next` will probably one of the hottest method call in codebase since we need to call it for each event.

Therefore, I believe that we should optimize it in a way that it should use minimum amount of memory and CPU. Specifically, I think we can provide two set of optimizations:

1. Add an extra `TrailDBEvent` parameter to `TrailDBCursor.next` and if it's not null, reuse that object instead of creating one. Avro is already using this method in order to minimize GC pressure: https://github.com/apache/avro/blob/17f2d75132021fafeca29edbdcade40df960fdc9/lang/java/avro/src/main/java/org/apache/avro/io/DatumReader.java#L35

2. Change the signature of `TrailDBCursor.next` to return `void` instead of `TrailDBEvent` and add `getCurrentItem()`, `getCurrentTimesamp()`, `getCurrentNumberOfItems()` to `TrailDBCursor`. We can store the reference of native `tdb_event` object inside `TrailDBCursor` and use the proxy methods instead of `TrailDBEvent`. The overhead of Java will be minimum in this case.

**Comment by aholyoke** [comment]
This is a good suggestion. TrailDBEvents get overwritten anyways due to the behavior of TrailDB. For example, 
```
cursor.getTrail(0);
TrailDBEvent e1 = cursor.next();
cursor.getTrail(1);
TrailDBEvent e2 = cursor.next();
System.out.println(e1.getItem(0));
```
Surprisingly, will print out the 0th item of e2. This is because a TrailDBEvent is essentially a pointer to the underlying `tdb_event` struct from the TrailDB api. And the TrailDB api reuses this struct between calls to `tdb_get_trail`.

**Comment by buremba** [comment]
I removed that part and implemented the reuse mechanism. The current signature or `next()` is similar to this one:
`JNIEXPORT jobject JNICALL Java_traildb_TrailDBCursor_next(JNIEnv *env, jobject obj, jlong db, jobject reuse)`

It was quite easy and I saw almost 3x performance improvement. If we can implement the method 2, it would be great since we can avoid sharing the objects between C and Java which is quite expensive. There is no `tdb_cursor_current` method in Traildb, any suggestions?
